website/static/js/wikiDiff.js

```diff
@@ -0,0 +1,38 @@
+var jsdiff = require('jsdiff');
+
+// Taken from diff.js
+function removeEmpty(array) {
+    var ret = [];
+    for (var i = 0; i < array.length; i++) {
+        if (array[i]) {
+            ret.push(array[i]);
+        }
+    }
+    return ret;
+}
+
+// Custom differ that includes newlines and sentence breaks
+var wikiDiff =  new jsdiff.Diff();
+wikiDiff.tokenize = function (value) {
+    return removeEmpty(value.split(/(\S.+?[.!?\n])(?=\s+|$)/));
+};
+
+var diff = function(beforeText, afterText) {
+    var diffList = wikiDiff.diff(beforeText, afterText);
+    var fragment = document.createDocumentFragment();
+
+    diffList.forEach(function(part) {
```

Comment:
Since performance is critical here, we might want to change this to a plain `for` loop, which performs faster (http://jsperf.com/foreach-vs-loop).